controllers/components/datasciencepipelines/datasciencepipelines_support.go

```diff
@@ -24,6 +24,7 @@ const (
 	// deployment to the new component name, so keep it around till we figure out a solution.
 	LegacyComponentName      = "data-science-pipelines-operator"
 	platformVersionParamsKey = "PLATFORMVERSION"
+	modelregistryParamsKey   = "MODELREGISTRY_NAMESPACE"
```

Comment:
We are making more and more logic based on the "model registry namespace" that implies the model registry instances are all in that namespace and not somewhere else.
This is not enforced as far as I remember and it is completely based on where the `ModelRegistry` CRD is created.

@dhirajsb 
Is this correct?
I think we should implement an admission webhook to enforce this aspect (at least in RHOAI).

Reply:
@danielezonca while the MR operator has no constraints about where an MR instance can be created, arch council prescribed the architecture of a shared MR namespace to consolidate MR specific configuration around servicemesh setup, etc. It also simplifies RBAC config by keeping it in a single namespace. 

Can you elaborate on how admission webhooks would work for this?

Reply:
> while the MR operator has no constraints about where an MR instance can be created, arch council prescribed the architecture of a shared MR namespace to consolidate MR specific configuration around servicemesh setup

Correct

My point is related to the scenario that you have described: there is no constraints to limit where the MR CR is created _but_ in reality we are making the assumption (strong, like here) that they are all in a specific namespace.
I'm suggesting to implement a [webhook](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook) (in RHOAI) to check the namespace of the MR CR resource during the creation and deny it if it is not in the namespace that we have decided.

Reply:
Just a point of clarification, the MR CR is cluster scoped but allows you to specify a namespace:
https://github.com/opendatahub-io/opendatahub-operator/blob/fcc8de99246b42e419779c60e48017d7847903cf/bundle/manifests/components.platform.opendatahub.io_modelregistries.yaml#L15

Reply:
@mprahl i think we are talking about this ModelReg https://github.com/opendatahub-io/model-registry-operator/blob/main/config/crd/bases/modelregistry.opendatahub.io_modelregistries.yaml not the modelregistries.components.platform.opendatahub.io

Reply:
Fir clarification, that CRD relates to an internal CRD, not the actual MR one

Reply:
@danielezonca @dhirajsb should we track this webhook suggestion as a JIRA ?

Reply:
Done [RHOAIENG-20962](https://issues.redhat.com/browse/RHOAIENG-20962)